js/components/siteInfo.js

```diff
@@ -16,18 +16,29 @@ class SiteInfo extends ImmutableComponent {
   constructor () {
     super()
     this.onAllowRunInsecureContent = this.onAllowRunInsecureContent.bind(this)
+    this.onDenyRunInsecureContent = this.onDenyRunInsecureContent.bind(this)
   }
   onAllowRunInsecureContent () {
-    appActions.changeSiteSetting(siteUtil.getOrigin(this.isBlockedRunInsecureContent), 'runInsecureContent', true)
+    appActions.changeSiteSetting(siteUtil.getOrigin(this.isBlockedRunInsecureContent),
+      'runInsecureContent', true, this.isPrivate)
     ipc.emit(messages.SHORTCUT_ACTIVE_FRAME_LOAD_URL, {}, this.isBlockedRunInsecureContent)
```

Comment:
the code above can also use `this.location` instead of `this.isBlockedRunInsecureContent`